client/src/components/Comments/index.tsx

```diff
@@ -1,34 +1,101 @@
-    {id: 5, image: noAvatar, username: 'Artem', text: 'If you have something advices, dont speak about this)))'},
-]
-
-export const Comment = (): JSX.Element => (
-    <div className='flex-grow-1 comments border-top position-relative'>
-        <div className='position-absolute h-100'>
-            { comments.map((comment: any) => (
-                <div className='one-comment px-3' key={comment.id}>
-                    <img
-                        src={comment.image || noAvatar}
-                        alt='avatar'
-                        width={24}
-                        height={24}
-                        className='img-fluid rounded-circle mt-1 mr-1 mb-1'
+interface ILocalState {
+    postId: string;
+    comments: IComment[];
+    commentsPage: number;
+    commentsLoading: boolean;
+    allCommentsLoaded: boolean;
+}
+
+interface IState {
+    comments: ILocalState;
+    ownProps: {
+        postId: string;
+    };
+}
+
+interface IOwnCommentsProps {
+    getComments: (postId: string) => void;
+    getMoreComments: (postId: string, page: number) => void;
+    addNextCommentsPage: (commentsPage: number) => void;
+    resetComments: () => void;
+}
+
+export type ICommentsProps = IOwnCommentsProps & ILocalState;
+
+class Comments extends React.Component<ICommentsProps> {
+    public componentDidMount(): void {
+        this.props.getComments(this.props.postId);
+    }
+
+    public componentWillUnmount(): void {
+        this.props.resetComments();
+    }
+
+    public getMoreComments = (): void => {
+        if (!this.props.allCommentsLoaded) {
+            this.props.addNextCommentsPage(this.props.commentsPage);
+            this.props.getMoreComments(this.props.postId, this.props.commentsPage);
+        }
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className='flex-grow-1 comments border-top position-relative'>
+                <div className='position-absolute h-100'>
+                    {!!this.props.comments && this.props.comments.map((comment: any) => (
+                        <div className='one-comment px-3' key={comment._id}>
+                            <img
+                                src={comment.authorId.photoPath || noAvatar}
+                                alt='avatar'
+                                width={24}
+                                height={24}
+                                className='img-fluid rounded-circle mt-1 mr-1 mb-1'
+                            />
+                            <span className='mt-1'>{comment.authorId.username}</span>
+                            <div className='d-inline-flex mt-3 float-right edit-delete-comment'>
+                                <i className='fa fa-pencil mr-2 edit-comment'/>
+                                <i className='fa fa-trash-o delete-comment'/>
+                            </div>
+                            <p>{comment.comment}</p>
```

Comment:
it seems to me that someone wrote component for it. Pls check

Reply:
Checked
No another component for it